greenpithumb/clock.py

```diff
@@ -43,12 +43,20 @@ def __init__(self, clock, duration):
         """
         self._clock = clock
         self._duration = duration
-        self._start_time = self._clock.now()
+        self.reset()
+
+    def set_remaining(self, time_remaining):
+        if time_remaining < datetime.timedelta(seconds=0):
+            raise ValueError('time_remaining must be non-negative')
```

Comment:
I want to be more defensive here because I feel like it's more likely that there's logic going on on the other side. For something like "make sure the parameter is not `None`" it's kind of a loud coding mistake on the caller's part. For something like this, the caller is probably isn't calling with something static like `timer.set_remaining(datetime.timedelta(minutes=5))`. They're probably doing some kind of arithmetic to calculate the time remaining.